### ISSUE TITLE
False positive E1136: Value 'typing.ChainMap' is unsubscriptable (unsubscriptable-object)

### ISSUE THREAD

**Issue by jhbuhrman** [PyCQA/pylint#2849]
### Steps to reproduce
1. Create the following file named `pylint_chainmap.py`:
```python
import typing
from collections import ChainMap


def check_pylint_chain_map() -> typing.ChainMap[str, typing.Any]:
    return ChainMap({"answer": 42})
```
2. `pylint --reports=n --score=n --disable=C0111 pylint_chainmap.py`

Please note that `mypy` is perfectly happy with this source file.

### Current behavior
```
************* Module pylint_chainmap
pylint_chainmap.py:5:32: E1136: Value 'typing.ChainMap' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior
```
```

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.6.6 (v3.6.6:4cf1f54eb7, Jun 27 2018, 03:37:03) [MSC v.1900 64 bit (AMD64)]
```

**Comment by jhbuhrman** [comment]
Might it have something to do with the conditional definition in `typing.py:2036`? ... 

```python
if hasattr(collections, 'ChainMap'):
    # ChainMap only exists in 3.3+
    __all__.append('ChainMap')

    class ChainMap(collections.ChainMap, MutableMapping[KT, VT],
                   extra=collections.ChainMap):
```

**Comment by PCManticore** [comment]
@jhbuhrman It's entirely possible that the inference is tripped by the conditional definition. We need to add support for some typing constructs in `astroid`, the library that `pylint` uses for type inference.
